src/tarski/syntax/terms.py

```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 from typing import Tuple
-
-from .sorts import Sort
+import numpy as np
```

Comment:
I would like to keep the term hierarchy as "pure" (i.e. close to the textbook definition) as possible. A term is either a variable or a compound term, i.e. a function symbol paired up with appropriate subterms (ok, we keep constants separate for the moment). 
I could accept an if-then-else, as it really doesn't fit the standard definition, I guess, because you cannot nest a formula in a term, etc. And I'm still in the process of understanding whether a Matrix and an Aggregate Compound Term really need to be distinct classes (see comments below).
But: in any case I would like to move these to some other file, to keep this file as readable as possible. In particular, I would like not to import numpy here, to keep things simple. What do you think, @miquelramirez? I could do that, just to get your opinion :-)

Reply:
See commit 84f69f367625a537f7e78cba01b57d97f8078eae for relocation of ```Matrix``` to somewhere else.